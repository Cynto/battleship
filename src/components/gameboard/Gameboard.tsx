import React from 'react'
import './gameboard.css'

function Gameboard(props: any) {
  return (
    <div data-testid="gameboard" className="gameboard-container">
      <h2></h2>
    </div>
  )
}

export default Gameboard
