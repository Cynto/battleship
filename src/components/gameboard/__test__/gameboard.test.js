import React from 'react';
import ReactDOM from 'react-dom';
import Gameboard from '../Gameboard';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';


describe('Start page tests', () => {
  afterEach(cleanup);
  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Gameboard />
      </Router>,
      div,
    );
  });
  test('Renders main container correctly', () => {
    const { getByTestId } = render(
      <Router>
        <Gameboard />
      </Router>,
    );
    expect(getByTestId('gameboard')).toHaveClass('gameboard-container');
  });
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Gameboard />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
