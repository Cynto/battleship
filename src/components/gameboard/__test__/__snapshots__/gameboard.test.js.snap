// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start page tests Matches snapshot 1`] = `
<div
  className="gameboard-container"
  data-testid="gameboard"
>
  <h2 />
</div>
`;
